hypervisor/common/hypercall.c

```diff
@@ -838,7 +838,7 @@ int32_t hcall_assign_ptdev(struct acrn_vm *vm, uint16_t vmid, uint64_t param)
 		if (vdev == NULL) {
 			pr_fatal("%s %x:%x.%x not found\n", __func__, bdf.bits.b,  bdf.bits.d,  bdf.bits.f);
 			ret = -EPERM;
-		} else {
+		} else if (vdev->pdev != NULL) {
```

Comment:
Then we need another else {}, for MISRA-C compliance.

Reply:
@dongyaozu  updated with combining the condition check for "vdev" and “vdev", and adding reviewers , thanks.